src/xcode/ENA/ENA/Source/Services/CCLService/DCCUIText.swift

```diff
@@ -0,0 +1,273 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import AudioToolbox
+import AnyCodable
+import jsonlogic
+
+enum ParameterType {
+	static let number = "number"
+	static let string = "string"
+	static let localDate = "localDate"
+	static let localDateTime = "localDateTime"
+	static let utcDate = "utcDate"
+	static let utcDateTime = "utcDateTime"
+}
+
+enum UITextType {
+	static let string = "string"
+	static let plural = "plural"
+	static let systemTimeDependent = "system-time-dependent"
+}
+
+struct DCCUITextParameter {
+	let type: String
+	let value: Any
+}
+
+public struct DCCUIText: Codable, Equatable {
+	let type: String
+	let quantity: Int?
+	let quantityParameterIndex: Int?
+	let functionName: String?
+	let localizedText: [String: AnyCodable]?
+	let parameters: AnyCodable
+
+	static let dateFormatter: ISO8601DateFormatter = .iso8601DateFormatter()
+	static let localDateFormatter: DateFormatter = .localDateFormatter()
+	static let localDateTimeFormatter: DateFormatter = .localDateTimeFormatter()
+	static let outputDateFormatter: DateFormatter = .outputDateFormatter()
+	static let outputDateTimeFormatter: DateFormatter = .outputDateTimeFormatter()
+	
+	// MARK: - Internal
+
+	func localized(languageCode: String? = Locale.current.languageCode) -> String? {
```

Comment:
Returning optional means, that there is no detailed error handling in place? What happens on the other side if nil is returned?

Reply:
Would it be possible to return an empty string instead of an optional?
This way we don't need to unwrap at every place we are using this API.